control-plane-operator/controllers/hostedcontrolplane/kas/egresscfg.go

```diff
@@ -0,0 +1,82 @@
+package kas
+
+import (
+	"bytes"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/serializer/json"
+	kasv1beta1 "k8s.io/apiserver/pkg/apis/apiserver/v1beta1"
+)
+
+func egressSelectorConfiguration() *kasv1beta1.EgressSelectorConfiguration {
+	return &kasv1beta1.EgressSelectorConfiguration{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "EgressSelectorConfiguration",
+			APIVersion: kasv1beta1.SchemeGroupVersion.String(),
+		},
+		EgressSelections: []kasv1beta1.EgressSelection{
+			{
+				Name: "controlplane",
+				Connection: kasv1beta1.Connection{
+					ProxyProtocol: kasv1beta1.ProtocolDirect,
+				},
+			},
+			{
+				Name: "etcd",
+				Connection: kasv1beta1.Connection{
+					ProxyProtocol: kasv1beta1.ProtocolDirect,
+				},
+			},
+			{
+				Name: "cluster",
+				Connection: kasv1beta1.Connection{
+					ProxyProtocol: kasv1beta1.ProtocolGRPC,
+					Transport: &kasv1beta1.Transport{
+						TCP: &kasv1beta1.TCPTransport{
+							URL:       "https://kconnectivity-server:8090",
+							TLSConfig: &kasv1beta1.TLSConfig{},
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+const (
+	EgressSelectorConfigMapKey = "config.yaml"
+)
+
+var (
+	kasScheme     = runtime.NewScheme()
+	kasSerializer = json.NewSerializerWithOptions(
+		json.DefaultMetaFactory, kasScheme, kasScheme,
+		json.SerializerOptions{Yaml: true, Pretty: true, Strict: true},
+	)
+)
+
+func init() {
+	kasv1beta1.AddToScheme(kasScheme)
+}
```

Comment:
We have a place for this stuff you can use: https://github.com/openshift/hypershift/blob/main/api/scheme.go#L36